### ISSUE TITLE
Parse error on variable named 'pattern'

### ISSUE THREAD

**Issue by mishun** [ndmitchell/hlint#216]
Example:
```Haskell
module Main where

f pattern = pattern

main = return ()
```
HLint v1.9.28 outputs:
```
 Error: Parse error: pattern
Found:
    module Main where
    
  > f pattern = pattern
    
    main = return ()
  

1 hint
```

PatternSynonyms are not enabled, so it is correct Haskell code.

**Comment by ndmitchell** [comment]
I believe this is fixed in haskell-src-exts HEAD, but not in the last released version. CC @mpickering.

**Comment by ndmitchell** [comment]
Passing `-XNoPatternSynonyms` is a temporary workaround.

**Comment by asr** [comment]
+1

**Comment by mpickering** [comment]
Yes, I think it is fixed but I'm not going to release a new version until GHC 8.0 is released.

**Comment by asr** [comment]
I don't see any *technical* reason for waiting for GHC 8.0.1. In addition, we could use the
release candidate for GHC 8.0.1 for testing the new version of HSE.

**Comment by mpickering** [comment]
I would release a version supporting 8.0 but I am waiting for Richard to document `TypeInType` so I can extend the parser however it needs to be to support that extension.

**Comment by ndmitchell** [comment]
Thanks @mpickering - that makes a lot of sense.

**Comment by ndmitchell** [comment]
Another reported instance at #236 - quickly becoming an FAQ.

**Comment by ndmitchell** [comment]
This is broken in GHC if you turn on PatternSynonyms, so passing -XNoPatternSynonyms is required.
